libqaul/common/src/lib.rs

```diff
@@ -1,9 +1,16 @@
 //! A small library with data types used across qaul.net
+extern crate blake2;
+mod message;
+mod payload;
+
+pub type HashBytes = [u8; 32];  // Because of the lack of type-level generics,
```

Comment:
I'm wondering if we maybe wanna use the [generic array](https://docs.rs/generic-array/0.13.0/generic_array/) crate here? I've never worked with it directly but a bunch of RustCrypto crates use it for exactly this reason

Reply:
If you're ok with making that our standard; it's hard to mix `GenericArray` with regular arrays.

Reply:
Yea, sounds good to me